lib/mayaUsd/ufe/UsdAttribute.cpp

```diff
@@ -477,18 +477,26 @@ std::string UsdAttributeFilename::get() const
         return std::string();
 
     PXR_NS::VtValue vt;
-    if (UsdAttribute::get(vt, getCurrentTime(sceneItem())) && vt.IsHolding<SdfAssetPath>()) {
-        SdfAssetPath path = vt.UncheckedGet<SdfAssetPath>();
-        return path.GetAssetPath();
+    if (UsdAttribute::get(vt, getCurrentTime(sceneItem()))) {
+        if (vt.IsHolding<SdfAssetPath>()) {
+            SdfAssetPath path = vt.UncheckedGet<SdfAssetPath>();
+            return path.GetAssetPath();
+        } else if (vt.IsHolding<std::string>()) {
+            return vt.UncheckedGet<std::string>();
+        }
     }
 
     return std::string();
 }
 
 void UsdAttributeFilename::set(const std::string& value)
 {
-    SdfAssetPath path(value);
-    setUsdAttr<PXR_NS::SdfAssetPath>(*this, path);
+    if (UsdAttribute::nativeType() == "string") {
```

Comment:
Ideally this should be the constant Ufe::Attribute::kString

Reply:
It would work, however nativeType is supposed to return a USD type as a string, so I have to use this longer constant that ends up having exactly the same value as Ufe::Attribute::kString.